File: flask/activities.py
from temporalio import activity


class SampleActivities:

    @activity.defn
    async def activity1(self) -> str:
        print(f"Activity 1")
        return "done"
    
    @activity.defn
    async def activity2(self) -> str:
        print(f"Activity 2")
        return "done"

    @activity.defn
    async def activity3(self, data: str |None = None) -> str:
        
        try:
            if data:
                return data
            raise Exception("Data is None")
        except Exception:
            activity.logger.exception("Refund failed")
            raise

    @activity.defn
    async def activity4(self) -> str:
        print(f"Activity 4")
        return "done"

File: server/src/temporal/activities.ts
export async function activity1() {
  console.log('activity1');
  return 'done';
}
export async function activity2() {
  console.log('activity2');
  return 'done';
}
export async function activity3(retry?: boolean) {
  console.log('activity3');
  if (!retry) {
    throw new Error('activity3 error');
  }
  return 'done';
}
export async function activity4() {
  console.log('Workflow completed');
  return 'done';
}
